aggregator/src/metrics.rs

```diff
@@ -225,3 +230,67 @@ pub fn install_metrics_exporter(
         None => Ok(MetricsExporterHandle::Noop),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    #[cfg(feature = "prometheus")]
+    use super::{
+        install_metrics_exporter, MetricsConfiguration, MetricsExporterConfiguration,
+        MetricsExporterHandle,
+    };
+    #[cfg(feature = "prometheus")]
+    use http::StatusCode;
+    #[cfg(feature = "prometheus")]
+    use janus_core::retries::{retry_http_request, test_http_request_exponential_backoff};
+    #[cfg(feature = "prometheus")]
+    use opentelemetry::Context;
+    #[cfg(feature = "prometheus")]
+    use std::net::Ipv4Addr;
+    #[cfg(feature = "prometheus")]
+    use tokio::net::TcpListener;
+
+    #[cfg(feature = "prometheus")]
+    #[tokio::test]
+    async fn prometheus_metrics_pull() {
+        let listener = TcpListener::bind((Ipv4Addr::LOCALHOST, 0)).await.unwrap();
+        let port = listener.local_addr().unwrap().port();
+        drop(listener);
```

Comment:
Is this an attempt to find an open port for this test's metrics listener to bind? Doing this way is racy. I think we should instead see if we can get Trillium to bind an ephemeral port. The documentation on on its [`Config`](https://docs.rs/trillium-server-common/latest/trillium_server_common/struct.Config.html#method.with_port) doesn't say you can do this, so maybe we need a new feature on Trillium? Or is there another Trillium testing affordance we can use?

Reply:
Yeah, though we are getting away with this race elsewhere in tests. I was able to get the port number from the `Info` argument passed to `Handler::init()` instead.

Reply:
I suppose there is still a race condition with portpicker though, but I seem to remember some issue with binding to 0 on windows

Reply:
That's alright, Janus CI doesn't build or test on anything but Linux x86_64, and so anything else is not supported.